### PR TITLE
Fix unmarshaling struct with interface field which is a pointer to another struct

### PR DESCRIPTION
**Pull request by rilinor**
This pull request fixes a bug connected with unmarshaling struct which contains interface field, where interface is a pointer to another struct. Instead of filled embedded struct we get `map[string]interface{}`. See `TestUnmarshalStructWithEmbeddedPtrStruct` for example.

### CODE REVIEW

**Comment by rilinor** [comment]
@vstarodub, what do you think about this pull request?

**Comment by rilinor** [comment]
Merge it please, we already use this in production, everything is good)

**Comment by savalin** [comment]
Please approve and merge this pull-request. This solution works well in production for 3 months.

**Comment by sh4d0w28** [comment]
Please, merge. Tested already in prod, works perfect.

**Comment by yurineydachin** [comment]
We really need it! Please, approve

**Comment by major28** [comment]
good solution ... please, merge

**Comment by o1egl** [comment]
I think that this is antipattern and transfer object should not contain interfaces.

**Comment by egpolyakov** [comment]
Really need it!

File: gen/generator.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ import (
 
 const pkgWriter = "github.com/mailru/easyjson/jwriter"
 const pkgLexer = "github.com/mailru/easyjson/jlexer"
-const pkgEasyjson = "github.com/mailru/easyjson"
+const pkgEasyJson = "github.com/mailru/easyjson"
```
Comment (bullgare) [inline gen/generator.go]: Why it's not in uppercase? I mean JSON

**Comment by mekegi** [comment]
Please don't merge this pull request

`-const pkgEasyjson = "github.com/mailru/easyjson"
 +const pkgEasyJson = "github.com/mailru/easyjson"`
this renaming is useless

File: gen/generator.go (hunk 1 above)
Comment (rilinor) [inline gen/generator.go]: thx, for your suggestion, i will fix it

**Comment by rilinor** [comment]
@o1egl in out project we use common struct for all responses:


    //easyjson:json
    type Response struct {
        success bool
        error error
        data interface{}
    }
and if i set specific struct which have easyjson marshal function for `data` field, instead of filled embedded struct I get `map[string]interface{}` in `data`. This logic differs from the standard `encoding/json` package. So this PR just makes marshal logic the same as standard `encoding/json` package.

**Comment by o1egl** [comment]
@rilinor you can use embedding and this will solve your issue.

`
type Response struct {
    success bool
    error error
}

type ConcreteResponse struct {
    Response
    data string
}
`

**Comment by rilinor** [comment]
@o1egl yes you are right, we can use this, but the main point of this PR makes easyjson lib more flexible and more similar to the standard `encoding/json` package to prevent confusing in this case.

**Comment by screwyprof** [comment]
This feature is both useful and helpful. Merge it please ASAP, as it provides a more natural way to deal with nested structures

**Pull request merged**